daemon/containerd/image_pull.go

```diff
@@ -68,6 +70,29 @@ func (i *ImageService) PullImage(ctx context.Context, image, tagOrDigest string,
 	infoHandler := snapshotters.AppendInfoHandlerWrapper(ref.String())
 	opts = append(opts, containerd.WithImageHandlerWrapper(infoHandler))
 
-	_, err = i.client.Pull(ctx, ref.String(), opts...)
+	img, err := i.client.Pull(ctx, ref.String(), opts...)
+	if err != nil {
+		return err
+	}
+
+	logger := logrus.WithFields(logrus.Fields{
+		"imageID": img.Target().Digest,
+		"tag":     ref.String(),
+	})
+	logger.Info("image pulled")
+
+	// The pull succeeded, check if we have a dangling image for this target
+	danglingImage, err := i.client.ImageService().Get(context.Background(), danglingImageName(img.Target().Digest))
```

Comment:
I think we can drop the check and just attempt to delete it.
If it doesn't exist, we can just ignore the NotExists error, and if it does, then it will be deleted :)

Reply:
❤️ thanks (and no real hurry)